lib/core/src/resource_administration.cpp

```diff
@@ -0,0 +1,266 @@
+
+        generalAdminInp_t input{};
+        input.arg0 = "add";
+        input.arg1 = "resource";
+        input.arg2 = _info.resource_name.data();
+        input.arg3 = _info.resource_type.data();
+        input.arg4 = vault_path.data();
+        input.arg5 = _info.context_string.data();
+        input.arg6 = _info.zone_name.data();
+
+        if (const auto ec = rxGeneralAdmin(&_comm, &input); ec < 0) {
+            return std::error_code{ec, std::generic_category()};
+        }
+
+        return {};
+    }
+
+    auto remove_resource(RxComm& _comm, const resource_name_type _name) -> std::error_code
+    {
+        generalAdminInp_t input{};
+        input.arg0 = "rm";
+        input.arg1 = "resource";
+        input.arg2 = _name.data();
+        input.arg3 = ""; // No --dryrun flag.
+
+        if (const auto ec = rxGeneralAdmin(&_comm, &input); ec < 0) {
+            return std::error_code{ec, std::generic_category()};
+        }
+
+        return {};
+    }
+
+    auto add_child(RxComm& _comm, const resource_name_type _parent, const resource_name_type _child) -> std::error_code
+    {
+        generalAdminInp_t input{};
+        input.arg0 = "add";
+        input.arg1 = "childtoresc";
+        input.arg2 = _parent.data();
+        input.arg3 = _child.data();
+        input.arg4 = ""; // Context string (cannot be null).
+
+        if (const auto ec = rxGeneralAdmin(&_comm, &input); ec < 0) {
+            return std::error_code{ec, std::generic_category()};
+        }
+
+        return {};
+    }
+
+    auto remove_child(RxComm& _comm, const resource_name_type _parent, const resource_name_type _child) -> std::error_code
+    {
+        generalAdminInp_t input{};
+        input.arg0 = "rm";
+        input.arg1 = "childfromresc";
+        input.arg2 = _parent.data();
+        input.arg3 = _child.data();
+
+        if (const auto ec = rxGeneralAdmin(&_comm, &input); ec < 0) {
+            return std::error_code{ec, std::generic_category()};
+        }
+
+        return {};
+    }
+
+    auto exists(RxComm& _comm, const resource_name_type _name) -> std::tuple<std::error_code, bool>
+    {
+        try {
+            const auto gql = fmt::format("select RESC_ID where RESC_NAME = '{}'", _name);
+            return {{}, query_builder{}.build(_comm, gql).size() > 0};
+        }
+        catch (...) {
+            return {std::error_code{SYS_UNKNOWN_ERROR, std::generic_category()}, false};
+        }
+    }
+
+    auto info(RxComm& _comm, const resource_name_type _name) -> std::tuple<std::error_code, std::optional<resource_info>>
+    {
+        const auto to_resource_status = [](const std::string& _status)
+        {
+            // clang-format off
+            if (_status == "up")   return resource_status::up;
+            if (_status == "down") return resource_status::down;
+            if (_status == "auto") return resource_status::auto_detect;
+            // clang-format on
```

Comment:
i know we're trying to get things consistent and easy to hold.   pretty sure the logic a bit deeper is just looking for the substring 'down', so will need to make sure that's considered here.

Reply:
Following our discussion, I will remove `auto` from the available status values.